fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -342,16 +376,15 @@ public boolean visit(String elementName, int elementIndex, Uri uri) {
                 throw invalidComboException(searchParamType, uri);
             }
 
-            // For REFERENCE search parameters, we need to treat Uris as tokens,
+            // For REFERENCE search parameters, we need to treat Uris as references,
             // not strings.
             if (REFERENCE.equals(this.searchParamType)) {
-                TokenParmVal p = new TokenParmVal();
-                p.setResourceType(resourceType);
-                p.setName(searchParamCode);
-                p.setUrl(searchParamUrl);
-                p.setVersion(searchParamVersion);
-                p.setValueCode(uri.getValue());
-                result.add(p);
+                try {
+                    result.add(buildReferenceParmVal(uri.getValue(), null));
+                } catch (FHIRSearchException e) {
+                    // Log the error, but skip it because we're not supposed to throw exceptions here
+                    log.log(Level.WARNING, "Error processing reference", e);
```

Comment:
Under which cases would we hit this exception?  Bad input data from clients?  Or only for bad code / bad IGs on the server end?

If it results from some bad/unexpected data from clients, maybe the log level should be reduced.

Reply:
ah, now i see it just matches what we were doing in the Reference visit method.  although now i'm wondering the same thing there...

Reply:
It's actually thrown from ReferenceUtil.getServiceBaseUrl() when a base URL is not valid. I believe this would only be from bad user data, but left as warning.

Reply:
OK, just something to watch for I guess